pkg/apis/core/validation/validation.go

```diff
@@ -6595,6 +6595,7 @@ func validateTopologySpreadConstraints(constraints []core.TopologySpreadConstrai
 			allErrs = append(allErrs, err)
 		}
 		allErrs = append(allErrs, validateMatchLabelKeys(subFldPath.Child("matchLabelKeys"), constraint.MatchLabelKeys, constraint.LabelSelector)...)
+		allErrs = append(allErrs, unversionedvalidation.ValidateLabelSelector(constraint.LabelSelector, subFldPath.Child("labelSelector"))...)
```

Comment:
We need to consider what happens to existing persisted data that doesn't pass this validation

This means we need to only apply this stricter validation when validating a create request or an update request of an object that already passes the strict validation.

The pattern we use for this is:
1. plumbing the PodValidationOptions struct down to this function
2. add an `AllowInvalidTopologySpreadConstraintLabelSelector bool` field to that struct
3. populate that field in `GetValidationOptionsFromPodSpecAndMeta`:
  * set it to false when creating a new pod
  * set it to true when updating a pod where the old pod has invalid topology spread constraint label selectors
  * set it to false otherwise
4. Make this new validation line conditional on that boolean field
5. Add unit tests:
  * validate a new pod (should not allow invalid selectors)
  * validate an updated pod where the old pod has invalid selectors (should allow the new pod to have an invalid selector)
  * validate an updated pod where the old pod has valid selectors (should not allow the new pod to have an invalid selector)

Reply:
thanks for review. PR has been updated. PTAL